ksql-engine/src/main/java/io/confluent/ksql/function/UdfCompiler.java

```diff
@@ -130,9 +133,14 @@ public UdfInvoker compile(final Method method, final ClassLoader loader) {
           scriptEvaluator.createFastEvaluator("return new " + generatedClassName
                   + "(args, returnType, metrics);",
               UdfArgSupplier.class, new String[]{"args", "returnType", "metrics"});
-      return evaluator.apply(Collections.singletonList(
-          SchemaUtil.getSchemaFromType(valueAndAggregateTypes.left)),
-          SchemaUtil.getSchemaFromType(valueAndAggregateTypes.right), metrics);
+
+      final List<Schema> args = Collections.singletonList(
+          SchemaUtil.getSchemaFromType(valueAndAggregateTypes.left));
+
+      final Schema returnValue = SchemaUtil.ensureOptional(
```

Comment:
Is there a test that ensures we "optionalize" this?

Reply:
I don't think its currently possible to have a non-optional type from a UDAF. However, I put the code here to ensure that remains the case.